src/Diagnostics.DataProviders/KustoClient.cs

```diff
@@ -138,8 +138,20 @@ private DataTableResponseObjectCollection ProcessKustoResponse(string responseCo
             {
                 throw new Exception($"Kusto query response exceeded the Kusto record count limit: {_queryLimitDocsLink}");
             }
-
-            return JsonConvert.DeserializeObject<DataTableResponseObjectCollection>(responseContent);
+            try
+            {
+                return JsonConvert.DeserializeObject<DataTableResponseObjectCollection>(responseContent);
+            }
+            catch (Newtonsoft.Json.JsonSerializationException ex)
```

Comment:
The above checks for "Kusto response exceeded record count limit" and "data size limit" are redundant with the try/catch - if this catch block is capable of recognizing those checks and returning the proper error, you can remove them.

Reply:
We are not sure right now if the kusto response in the "response exceeded record count limit" follows the same exception structure. Will look further if this can be generalized.